src/applications/edu-benefits/components/AutosuggestField.jsx

```diff
@@ -0,0 +1,296 @@
+    if (getOptions) {
+      getOptions(inputValue)
+        .then(({ options, searchTerm }) =>
+          // match response to current request
+          searchTerm === this.state.searchTerm &&
+          this.setOptions(options));
+    }
+  }
+
+  setOptions = (options) => {
+    if (!this.unmounted) {
+      const optionsOrDefault = options.length > 0 ?
+        options :
+        [{ value: null, label: 'No results' }];
+      this.setState({
+        loading: false,
+        searchTerm: '',
+        options: optionsOrDefault,
+        suggestions: this.getSuggestions(optionsOrDefault, this.state.input)
+      });
+    }
+  }
+
+  getSuggestions = (options, value) => {
+    if (value) {
+      const uiOptions = this.props.uiSchema['ui:options'];
+      const sortedOptions = sortListByFuzzyMatch(value, options).slice(0, uiOptions.maxOptions);
+      return sortedOptions;
+    }
+
+    return options;
+  }
+
+  getFormData = (suggestion) => {
+    if (this.useEnum) {
+      return suggestion.id;
+    }
+
+    // When freeInput is true, we'll return the label to the api instead of the id
+    if (this.props.uiSchema['ui:options'].freeInput) {
+      return suggestion.label;
+    }
+
+    return _.set('widget', 'autosuggest', suggestion);
+  }
+
+  handleClearSelection = (e, downshiftClear) => {
+    e.preventDefault();
+    // using Downshift's clear sets the correct focus state after using clear
+    downshiftClear(() => this.setState({
+      input: '',
+      searchTerm: '',
+      loading: false
+    }));
+  }
+
+  handleInputValueChange = (inputValue) => {
+    if (inputValue !== this.state.input) {
+      const uiOptions = this.props.uiSchema['ui:options'];
+      if (uiOptions.queryForResults) {
+        this.setState({
+          loading: true,
+          searchTerm: inputValue
+        });
+        this.debouncedGetOptions(inputValue);
+      }
+
+      let item = { widget: 'autosuggest', label: inputValue };
+      // once the input is long enough, check for exactly matching strings so that we don't
+      // force a user to click on an item when they've typed an exact match of a label
+      if (inputValue && inputValue.length > 3) {
+        const matchingItem = this.state.suggestions.find(suggestion => suggestion.label === inputValue);
+        if (matchingItem) {
+          item = this.getFormData(matchingItem);
+        }
+      }
+
+      this.props.onChange((this.props.uiSchema['ui:options'].freeInput || this.useEnum) ? inputValue : item);
+      this.setState({
+        input: inputValue,
+        suggestions: this.getSuggestions(this.state.options, inputValue)
+      });
+    } else if (inputValue === '') {
+      this.props.onChange();
+      this.setState({
+        input: inputValue,
+        suggestions: this.getSuggestions(this.state.options, inputValue)
+      });
+    }
+  }
+
+  handleChange = (selectedItem) => {
+    const value = this.getFormData(selectedItem);
+    this.props.onChange(value);
```

Comment:
This component will need to call something to fetch the details about the institution during this event. Once those details are fetched, they need to populate the other form fields. 

My assumption is that the form config will pass a callback (e.g. on `ui:options`) to fetch the data and return an object that gets merged onto the existing `formData` _via_ this prop. I don't think there's away in the config to do this easily.